pkg/proxy/iptables_portalmanager.go

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
```

Comment:
Understandable :)

I think I'm going to go update my GSoC proposal to reflect trying to add pf based on this abstraction or similar and start tinkering with that, but I can do that in a separate branch or something.